core/io/resource_loader.cpp

```diff
@@ -454,6 +454,22 @@ Ref<Resource> ResourceLoader::load(const String &p_path, const String &p_type_hi
 	return res;
 }
 
+Ref<Resource> ResourceLoader::get_shallow_resource(const String &p_path, Error *r_error) {
+	if (r_error) {
+		*r_error = OK;
+	}
+
+	String local_path = _validate_local_path(p_path);
+	{
+		MutexLock thread_load_lock(thread_load_mutex);
+		if (!thread_load_tasks.has(local_path)) {
+			*r_error = FAILED;
```

Comment:
```suggestion
			if (r_error) {
				*r_error = FAILED;
			}
```

Reply:
I changed my mind for the method signature, the error reference will be required, as it is crucial to know if the `nullptr` returned has succeeded or failed.

I'll add a error check instead.